### ISSUE TITLE
Add test to validate broken links in markdown files

### ISSUE THREAD

**Issue by wearyzen** [firecracker-microvm/firecracker#4342]
We have lot of links in our md files which are broken and could be broken due to refactoring of code.
We need to add a new CI test which checked for all the links in all the md files in the repo.
The discussion started with https://github.com/firecracker-microvm/firecracker/pull/4341#discussion_r1435022112
and we found [mlc](https://crates.io/crates/mlc) which can validate all links in the md files however it fails to validate links that are embedded like https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md?plain=1#L79

The new test should cover links in all md files (embedded in code or not).

**Comment by andr3wy** [comment]
Hi everyone,

We're a group at UT Austin and we're interested in picking this issue up.

**Comment by roypat** [comment]
Hey all, 
We added a test to ensure that inter-markdown links are not broken in #4421. We decided against adding something that also validates external links to our CI for multiple reasons:
1. We do not want to send a bunch of HTTP requests to random websites every time a PR gets updated
2. Events external to this repository could cause the test to fail (say a temporary outage of a linked website), causing such a test to fail intermittently
3. The `mlc` tool above has a lot of false positives (for example regarding images embedded in markdown).

For these reasons, we are closing this issue
